app/services/analytics.rb

```diff
@@ -95,12 +96,44 @@ def browser_attributes
   end
 
   def session_duration
-    @session[:session_started_at].present? ? Time.zone.now - session_started_at : nil
+    session[:session_started_at].present? ? Time.zone.now - session_started_at : nil
   end
 
   def session_started_at
-    value = @session[:session_started_at]
+    value = session[:session_started_at]
     return value unless value.is_a?(String)
     Time.zone.parse(value)
   end
+
+  def sp_request_attributes
+    resolved_result = resolved_authn_context_result
+    return if resolved_result.nil?
+
+    attributes = resolved_result.to_h
+    attributes[:component_values] = resolved_result.component_values.map do |v|
+      v.to_h.slice(:name)
+    end
```

Comment:
```suggestion
    attributes[:component_values] = resolved_result.component_values.map do |v|
      [v.name, true]
    end.to_h
```
I really like [Matt's suggestion](https://github.com/18F/identity-idp/pull/10265#discussion_r1530908079) of making the components the keys because it makes it much easier to query for in Cloudwatch

Reply:
And in 012036826 I take your much cleaner implementation of this! Thanks!